src/sqlfluff/rules/L026.py

```diff
@@ -148,7 +148,8 @@ def _should_ignore_reference(reference, selectable):
         # - They are the target table, similar to an INSERT or UPDATE
         #   statement, thus not expected to match a table in the FROM
         #   clause.
-        return any(seg.is_type("into_table_clause") for seg in ref_path)
+        if ref_path:
```

Comment:
We should add an `else` here. This function is expected to return `True` or `False`, and now it may just implicitly exit (thus returning `None`).

Reply:
Done